python/soma/spm/spm12/tools/shoot_tools/__init__.py

```diff
@@ -47,3 +49,143 @@ def _move_output_default_path(self, output_list, output_prefix):
+        the total amount of signal in the images. Areas that are expanded
+        during warping are correspondingly reduced in intensity.
+        """
+        self.preserve = 1
+
+    def setFWHM(self, fwhm):
+        """
+        Specify the full-width at half maximum (FWHM) of the isotropic
+        Gaussian blurring kernel in mm.
+        """
+        self.fwhm = fwhm
+    
+    @checkIfArgumentTypeIsStrOrUnicode(argument_index=1)
+    def setDeformationFieldPath(self, deformation_field_path):
+        """
+        Select the deformation field generated by Shoot).
+        """
+        self.deformation_field_path = deformation_field_path
+    
+    @checkIfArgumentTypeIsAllowed(list, 1)
+    def setImagesPathList(self, images_path_list):
+        """
+        The deformation can be applied to multiple images.
+        Select images to spatially normalise.
+        """
+        self.images_path_list = images_path_list
+
+    @checkIfArgumentTypeIsAllowed(list, 1)
+    def setOutputImagesPathList(self, output_images_path_list):
+        self.output_images_path_list = output_images_path_list
+
+    def getStringListForBatch(self):
+        if not None in [self.deformation_field_path, self.images_path_list]:
+            batch_list = []
+            #input files
+            batch_list.append("spm.tools.shoot.norm.template = {'%s'};" % self.final_template_path)
+            batch_list.append("spm.tools.shoot.norm.data.subj.deformation = {'%s'};" % self.deformation_field_path)
+            batch_list.append("spm.tools.shoot.norm.data.subj.images = {%s};"
+                              % convertPathListToSPMBatchString(self.images_path_list, add_dimension=False))
+            #options
+            batch_list.append("spm.tools.shoot.norm.vox = %s;" %convertlistToSPMString(self.voxel_sizes))
+            batch_list.append("spm.tools.shoot.norm.bb = %s;" %convertNumpyArrayToSPMString(self.bounding_box))
+            batch_list.append("spm.tools.shoot.norm.preserve = %i;" %self.preserve)
+            batch_list.append("spm.tools.shoot.norm.fwhm = %s;" %self.fwhm)
+            return batch_list
+        else:
+            raise ValueError("deformation_field_path and images_path_list are required")
+
+    def _moveSPMDefaultPathsIfNeeded(self):
+        smoothing = (self.fwhm != 0)
+        if self.output_images_path_list:
+            for image_path, output_image_path in zip(self.images_path_list, self.output_images_path_list):
+                if smoothing and self.preserve:
+                    prefix = "smw"
+                elif smoothing and not self.preserve:
+                    prefix = "sw"
+                elif self.preserve:
+                    prefix = "mw"
+                else:
+                    prefix = "w"
+                moveSPMPath(image_path,
+                            output_image_path,
+                            prefix=prefix)
+            else:
+                pass #do not move default spm paths
```

Comment:
The `else` is aligned with the for, I think that the aim of it is to be the `else` of `if self.output_images_path_list:`.
As nothing is made inside this else, it will not change the behavior of the function. Maybe we can remove it?

Reply:
Yes it is suppose to be aligned with the first if :). I took the code from Mickaël but yes in fact it does nothing. If you prefer we can remove it and I will reformulate the comment to put it in the beginning.